src/main/java/org/jasig/portlet/newsreader/processor/RomeNewsProcessorImpl.java

```diff
@@ -38,47 +47,42 @@
 import org.owasp.validator.html.ScanException;
 import org.springframework.core.io.Resource;
 
-import com.sun.syndication.feed.module.Module;
-import com.sun.syndication.feed.module.mediarss.MediaEntryModule;
-import com.sun.syndication.feed.module.mediarss.types.MediaContent;
-import com.sun.syndication.feed.module.mediarss.types.MediaGroup;
-import com.sun.syndication.feed.synd.SyndContent;
-import com.sun.syndication.feed.synd.SyndEnclosure;
-import com.sun.syndication.feed.synd.SyndEntry;
-import com.sun.syndication.feed.synd.SyndFeed;
-import com.sun.syndication.io.FeedException;
-import com.sun.syndication.io.SyndFeedInput;
-import com.sun.syndication.io.XmlReader;
-
 public class RomeNewsProcessorImpl {
     
     protected final Log log = LogFactory.getLog(getClass());
 
     private List<String> imageTypes;
 
+    private int ENTRIES_PER_PAGE = 10;
+    
+    public void setEntriesPerPage(int perPage) {
+        this.ENTRIES_PER_PAGE = perPage;
+    }
+    
     public void setImageTypes(List<String> imageTypes) {
         this.imageTypes = imageTypes;
     }
-    
+
     private List<String> videoTypes;
 
     public void setVideoTypes(List<String> videoTypes) {
         this.videoTypes = videoTypes;
     }
 
-    public NewsFeed getFeed(InputStream in, String titlePolicy, String descriptionPolicy) throws IOException, IllegalArgumentException, FeedException, PolicyException, ScanException {
+    public PaginatingNewsFeed getFeed(InputStream in, String titlePolicy, String descriptionPolicy) throws IOException, IllegalArgumentException, FeedException, PolicyException, ScanException {
         // get a vanilla SyndFeed from the input stream
         XmlReader reader = new XmlReader(in);
         SyndFeedInput input = new SyndFeedInput();
         SyndFeed feed = input.build(reader);
 
-        NewsFeed newsFeed = new NewsFeed();
+        PaginatingNewsFeed newsFeed = new PaginatingNewsFeed(ENTRIES_PER_PAGE);
         newsFeed.setAuthor(feed.getAuthor());
         newsFeed.setLink(feed.getLink());
         newsFeed.setTitle(feed.getTitle());
         newsFeed.setCopyright(feed.getCopyright());
-        
-        List<NewsFeedItem> newEntries = new ArrayList<NewsFeedItem>();
+
+//        List<NewsFeedItem> newEntries = new ArrayList<NewsFeedItem>();
+        List<NewsFeedItem> newEntries = newsFeed.getEntries();
```

Comment:
It's a bit weird to manipulate the contents of another class' member variable outside that class.